File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/builder/vechicle/PlaneBuilder.java
package com.bigboxcode.designpattern.builder.vechicle;

public class PlaneBuilder implements VehicleBuilder {
    private int wheel;
    private int engine;
    private int seat;

    private boolean interior;
    private int door;
    private int wing;

    @Override
    public void addWheel(int noOfWheel) {
        System.out.println("Add " + noOfWheel + " wheels");

        this.wheel += noOfWheel;
    }

    @Override
    public void addEngine(int noOfEngine) {
        System.out.println("Add " + noOfEngine + " engine");

        this.engine += noOfEngine;
    }

    @Override
    public void addSeat(int noOfSeat) {
        System.out.println("Add " + noOfSeat + " Seat");

        this.seat = noOfSeat;
    }

    @Override
    public void addInterior() {
        System.out.println("Add interior");

        this.interior = true;
    }

    @Override
    public void addDoor(int noOfDoor) {
        System.out.println("Add " + noOfDoor + " door");

        this.door += noOfDoor;
    }

    @Override
    public void addWing(int noOfWings) throws Exception {
        System.out.println("Add " + noOfWings + " wing");

        this.wing += wing;
    }

    public Plane build() {
        Plane plane = new Plane(wheel, engine, seat, door, wing, interior);

        return plane;
    }

}

File: design-patterns/php/builder/vehicle/PlaneBuilder.php
<?php
// PlaneBuilder.php

namespace Builder\Vehicle;

class PlaneBuilder implements VehicleBuilder {
    private int $wheel = 0;
    private int $engine = 0;
    private int $seat = 0;
    private bool $interior;
    private int $door = 0;
    private int $wing = 0;

    public function addWheel(int $noOfWheel): void {
        echo "Add " . $noOfWheel . " wheels\n";

        $this->wheel += $noOfWheel;
    }

    public function addEngine(int $noOfEngine): void {
        echo "Add " . $noOfEngine . " engine\n";

        $this->engine += $noOfEngine;
    }

    public function addSeat(int $noOfSeat): void {
        echo "Add " . $noOfSeat . " Seat\n";

        $this->seat = $noOfSeat;
    }

    public function addInterior(): void {
        echo "Add interior\n";
        $this->interior = true;
    }

    public function addDoor(int $noOfDoor): void {
        echo "Add " . $noOfDoor . " door\n";

        $this->door += $noOfDoor;
    }

    public function addWing(int $noOfWing): void {
        echo "Add " . $noOfWing . " wing\n";

        $this->wing += $noOfWing;
    }

    public function build(): Plane {
        return new Plane($this->wheel, $this->engine, $this->seat, $this->door, $this->wing, $this->interior);
    }
}